tree: add -C, --du and --prune examples (#4332)
tldr-pages/tldr: pages/linux/tree.md

`tree -d`

- Print hidden files too with colorization on:

`tree -a -C`

- Print the tree without indentation lines, showing the full path instead (use `-N` to not escape whitespace and special characters):

`tree -i -f`

> - Print the size of each node next to it, in human-readable format, with directories displaying their cumulative size (as in the `du` command):

Reviewer: Wow, this is a pretty long description. Is this syncing with another copy of this page, or is it original?

If the latter, then if we could shorten this a bit and make it more concise that'd be great :-)

Same goes for the other particularly long one further down.

Author: This commit extended the description https://github.com/tldr-pages/tldr/commit/c294bdc4 in September 2016.

Author: @sbrl This commit extended the description https://github.com/tldr-pages/tldr/commit/c294bdc4 in September 2016.

Reviewer 2: I would simplify it somewhere around this:

```suggestion
- Print the size of each file and the cumulative size of each directory, in human-readable format:
```
What do you think?

`tree -s -h --du`

- Print files within the tree hierarchy, using a wildcard (glob) pattern, and pruning out directories that don't contain matching files:

`tree -P '{{*.txt}}' --prune`

> - Print directories within the tree hierarchy, using the wildcard (glob) pattern, and pruning out directories that aren't ancestors of the wanted one:

Reviewer 2: I understand you wanting to add more command line options, to this page, but this is rather advanced usage. Think of it this way: is this in the top 8 most common usages of `tree` you can think of? I would suggest keeping the original, or at best elaborating it while still keeping it simple.

Author: I'm just syncing this page with https://github.com/navarroaxel/tldr/blob/master/pages/osx/tree.md
Do you think we should shrink the osx page too?

Reviewer 2: @navarroaxel okay, that's a good point, thank you for linking. I see that the OSX page has different because the command actually supports different options (see [relevant commit](https://github.com/tldr-pages/tldr/commit/31a07858ca2d010625499fb9e942f9c1371b339a) where this was introduced). Is this example compatible with the Linux version of the tool? If yes, then you are right that it makes sense to sync them, and we can work on a PR to simplify both later.

Author: Yeah, I've checked and the same options applies to the Linux `tree`, and all these arguments are explained in the `more information` link in this page.

Author: @mebeim Yeah, I've checked and the same options applies to the Linux `tree`, and all these arguments are explained in the `more information` link in this page, the same link than `osx` page.

Reviewer 2: @navarroaxel thanks a lot
